UI/window-main.hpp

```diff
@@ -16,3 +17,138 @@ class OBSMainWindow : public QMainWindow {
 	virtual int GetProfilePath(char *path, size_t size, const char *file)
 		const=0;
 };
+
+class QHotkeyEvent : public QEvent {
+	obs_key_t _key;
+	uint32_t _modifiers;
+	bool _pressed;
+public:
+	static const QEvent::Type hotkeyType = static_cast<QEvent::Type>(QEvent::User + 0x4B45);
+	QHotkeyEvent(obs_key_t key, bool pressed, uint32_t modifiers = 0) :
+		QEvent(hotkeyType)
+	{
+		_key = key;
+		_pressed = pressed;
+		_modifiers = modifiers;
+	}
+	QHotkeyEvent(void* data) :
+		QEvent(hotkeyType)
+	{
+		UNUSED_PARAMETER(data);
+		_key = OBS_KEY_NONE;
+		_pressed = false;
+		_modifiers = 0;
+	}
+	virtual obs_key_t getKey()
+	{
+		return _key;
+	}
+	virtual bool pressed()
+	{
+		return _pressed;
+	}
+	virtual uint32_t getModifiers()
+	{
+		return _modifiers;
+	}
+};
+
+class QMidiEvent : public QHotkeyEvent {
+	std::vector<uint8_t> _message;
+	double _deltaTime;
+public:
+	//static const QEvent::Type midiType = static_cast<QEvent::Type>(QEvent::User + 0x4D49);
+	QMidiEvent(std::vector<uint8_t> message, double deltatime) :
+			QHotkeyEvent(nullptr)
```

Comment:
This is tabbed twice while the similar lines above it are tabbed once.  Find the appropriate tab style, and use that consistently.